arch/riscv/Kconfig

```diff
@@ -338,6 +338,14 @@ config GEN_IRQ_VECTOR_TABLE
 config ARCH_HAS_SINGLE_THREAD_SUPPORT
 	default y if !SMP
 
+config CBOM_ICACHE_SUPPORTED
+	bool
+	depends on RISCV_ISA_EXT_ZICBOM && ICACHE
+
+config CBOM_DCACHE_SUPPORTED
+	bool
+	depends on RISCV_ISA_EXT_ZICBOM && DCACHE
```

Comment:
Yes, I think !RISCV_ISA_EXT_ZICBOM didn't suffice enought, at less, in the case of Andes cores, there is a situation where only Data Cache (D cache) CMO is supported, and Instruction Cache (I cache) CMO is not supported.

I need to update my perspective. I believe that CBOM_{I/D}CACHE_SUPPORTED is still necessary. The reason is that even if a SoC does not select CBOM_{I/D}CACHE_SUPPORTED, it may still need to select CONFIG_CPU_HAS_{I/D}CACHE. It will operate on the I/D cache based on the existence of CONFIG_CPU_HAS_{I/D}CACHE(without CMO extension), such as enable and setting I/D cache in the start code during the pre-kernel.